FinderCompass/src/main/java/atomicstryker/findercompass/common/CompassTargetData.java

```diff
@@ -29,14 +64,53 @@ public boolean equals(Object o)
         if (o instanceof CompassTargetData)
         {
             CompassTargetData comp = (CompassTargetData)o;
-            return comp.getBlockID() == blockID && comp.getDamage() == damage;
+            return comp.getOreDictName().equals(mOreDictName) ;
         }
         return false;
     }
     
     @Override
     public int hashCode()
     {
-        return blockID.getUnlocalizedName().hashCode();
+        return mOreDictName.hashCode();
+    }
+    
+    public class BlockData{
+        private final Block blockID;
+        private final int damage;
+        
+        private BlockData(Block pBlock, int pDamage){
+        	blockID = pBlock;
+        	damage = pDamage;
+        }
+        
+        public Block getBlockID()
+        {
+            return blockID;
+        }
+        
+        public int getDamage()
+        {
+            return damage;
+        }
+
+        
+        @Override
+        public boolean equals(Object o)
+        {
+            if (o instanceof BlockData)
+            {
+            	BlockData comp = (BlockData)o;
+                return comp.getBlockID() == blockID && comp.getDamage() == damage;
+            }
+            return false;
+        }
+        
+        @Override
+        public int hashCode()
+        {
+            return blockID.getUnlocalizedName().hashCode();

```

Comment:
I'm not sure, if it is possible or not to get a null return. I'll check the source in forge and optional add a second function.